contracts/core/MarginRequirements.sol

```diff
@@ -0,0 +1,230 @@
+        require(
+            _initialMargin > 0 && _initialMargin <= 100 * 10**2,
+            "MarginRequirements: initial margin cannot be 0 or higher than 100%"
+        );
+        require(_underlying != address(0), "MarginRequirements: invalid underlying");
+        require(_collateralAsset != address(0), "MarginRequirements: invalid collateral");
+        require(_account != address(0), "MarginRequirements: invalid account");
+
+        initialMargin[keccak256(abi.encode(_underlying, _collateralAsset, _isPut))][_account] = _initialMargin;
+    }
+
+    /**
+     * @notice sets the maintenance margin absolute amount
+     * @dev can only be called by keeper
+     * @param _vaultID id of the vault
+     * @param _maintenanceMargin maintenance margin absolute amount with its respective token decimals
+     */
+    function setMaintenanceMargin(uint256 _vaultID, uint256 _maintenanceMargin) external onlyKeeper {
+        require(_maintenanceMargin > 0, "MarginRequirements: initial margin cannot be 0");
+
+        maintenanceMargin[_vaultID] = _maintenanceMargin;
+    }
+
+    /************************************************
+     *  MARGIN OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice checks if there is enough collateral to mint the desired amount of otokens
+     * @param _account account address
+     * @param _notional order notional amount (USD value with 6 decimals)
+     * @param _underlying underlying asset address
+     * @param _isPut option type the vault is selling
+     * @param _collateralAsset collateral asset address
+     * @param _collateralAmount collateral amount (with its respective token decimals)
+     * @return boolean value stating whether there is enough collateral to mint
+     */
+    function checkMintCollateral(
+        address _account,
+        uint256 _notional,
+        address _underlying,
+        bool _isPut,
+        uint256 _collateralAmount,
+        address _collateralAsset
+    ) external view returns (bool) {
+        // retrieve collateral decimals
+        uint256 collateralDecimals = uint256(ERC20Interface(_collateralAsset).decimals());
+
+        // retrieve initial margin
+        uint256 initialMarginRequired = initialMargin[keccak256(abi.encode(_underlying, _collateralAsset, _isPut))][
+            _account
+        ];
+
+        // InitialMargin < Collateral
+
+        // Starts with:
+        // notional (USD) * (initial margin/100) < collateral (#tokens) * collateral price (in USD)
+
+        // initial margin is dividing by 100 since it is a %. Then, 100 moves to the other equation side multiplying:
+        // notional (USD) * initial margin < collateral (#tokens) * collateral price * 100
+
+        // Remaining values are added to ensure both sides of the equation are scaled equally given they differ in decimal amounts
+
+        return
+            _notional.mul(initialMarginRequired).mul(10**collateralDecimals).mul(10**ORACLE_DECIMALS) <
```

Comment:
lets make it <=

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/43dc6a1c7b5ae7ca86b7385ffef1f2249e1ab9c0